app/src/main/java/com/cp/tasky/auth/login/presentation/LoginViewModel.kt

```diff
@@ -46,18 +45,18 @@ class LoginViewModel @Inject constructor(
     }
 
     private fun loginUser(email: String, password: String) {
-        _networkState.value = Result.Loading
-
         viewModelScope.launch {
+            _networkResponse.send(Result.Loading)
```

Comment:
Everything that is a state (something you want to persist, even across screen rotations) should always stay state. Therefore, you wouldn't want to send the loading state into a channel, but rather update a field of your StateFlow with it. 
Use the channel only if you have one-time events you just want to handle exactly once across configuration changes, like showing a Toast or navigating. Most other things are handled with persistent state.